#163388100 Get Articles Feature

asirvex opened the pull request:
#### What does this PR do?
It enables a user to create an article
#### Description of Task to be completed?

 - Create the list article component
 - Add the functionality for list articles
 - Create the get an article component
 - Add the functionality for getting an article


#### How should this be manually tested?
1. Clone the repo
```
git clone https://github.com/andela/ah-the-phoenix-frontend.git
```
2. Change it into the current directory
```
cd ah-the-phoenix-frontend
```
3. Checkout to the feature branch
```    
git checkout ft-get-article-163388100
```
4. Install the required dependencies using 
```
npm install
```
5. Run the server
```
npm start
```
6. Login to the website then go to the following urls
```
/articles
/articles/<article-slug>
```

Any background context you want to provide?
N/A

What are the relevant pivotal tracker stories?
[#163388100](https://www.pivotaltracker.com/story/show/163388100)

ezrqnkemboi commented:
Squash one commit @asirvex.

ezrqnkemboi commented on package.json:
```diff
@@ -7,13 +7,14 @@
     "eslint-plugin-jsx-a11y": "^6.2.1",
-    "axios": "^0.18.0",
     "firebase": "^5.8.6",
+    "lorem-ipsum": "^2.0.0",
```
Why are we using this?

ezrqnkemboi commented on src/components/Articles/Article.js:
```diff
@@ -0,0 +1,56 @@
+    componentDidMount() {
+        this
+            .props
+            .getArticle()
```
This is not good practice. 
It should be `this.props.getArticle()` in one line, for good readability.

asirvex replied:
Okay, let me correct that. The beautify plugin changed it to that on save

asirvex committed:
Feature(get articles): a user should get articles

- create the component for the listing articles
- create component for displaying an article
- create actions and reducers for getting articles
- create actions and reducers for getting a single article

[Finish #163388100]

ezrqnkemboi commented on src/components/Articles/ListArticles.js:
```diff
@@ -0,0 +1,87 @@
+export class ListArticles extends Component {
+
+    componentDidMount = () => {
+        this
```
Make this readable.

ezrqnkemboi commented on src/components/Articles/ListArticles.js:
```diff
@@ -0,0 +1,87 @@
+
+    getWords = str => {
+        return str
+            .split(/\s+/)
```
We can make this better too.

Arusey commented:
@asirvex consider getting articles on the profiles route. It allows users to see what articles have been created by the current user

ezrqnkemboi commented:
@asirvex, have you worked on @Arusey suggestion or have you had sync with him to get more clarification?

ezrqnkemboi commented on src/redux/actiontypes.js:
```diff
@@ -30,3 +30,11 @@ export const PROFILE_FAILURE = 'PROFILE_FAILURE';
+export const GET_ARTICLES_SUCCESS = 'GET_ARTICLES_SUCCESS';
+export const GET_ARTICLES_FAILURE = 'GET_ARTICLES_FAILURE';
+
+export const GET_ARTICLE_REQUEST = 'GET_ARTICLE_REQUEST';
```
Rename ```GET_SINGLE_ARTICLE_REQUEST```
